### DISCUSSION TITLE
Battery state of charge

### DISCUSSION THREAD

**Discussion by eismarkt** [johanmeijer/grott#294] [General]
I am missing the battery state of charge from the hybrid inverter Sph 10000 Bh up.
Where can I find it

**Comment by johanmeijer** [comment]
Can you see which layout is beeing used (eg T060104XSPF or T050104XSPF)?

**Comment by eismarkt** [comment]
i think both inverters are using the same T06NNNNX

- Possible Inverter serial RCL1*****
	 - Inverter serial not recognised - using inverter type default
	 - Growatt new layout processing
		 - decrypt       :  True
		 - offset        :  6
		 - record layout :  T06NNNNX
	 - Grott data record date/time processing started
	 - no or no valid time/date found, grott server time will be used (buffer records not sent!)
	 - Grott values retrieved:
		 -  datalogserial        :  KWK1C****
		 -  pvserial             :  RCL1C****
		 -  pvstatus             :  1
		 -  pvpowerin            :  369.7
		 -  pv1voltage           :  288.0
		 -  pv1current           :  1.1
		 -  pv1watt              :  369.6
		 -  pv2voltage           :  4.2
		 -  pv2current           :  0.0
		 -  pv2watt              :  0.1
		 -  pvpowerout           :  362.5
		 -  pvfrequentie         :  50.0
		 -  pvgridvoltage        :  233.4
		 -  pvgridcurrent        :  1.5
		 -  pvgridpower          :  363.7
		 -  pvgridvoltage2       :  0.0
		 -  pvgridcurrent2       :  0.0
		 -  pvgridpower2         :  0.0
		 -  pvgridvoltage3       :  0.0
		 -  pvgridcurrent3       :  0.0
		 -  pvgridpower3         :  0.0
		 -  totworktime          :  346.1
		 -  pvenergytoday        :  2.4
		 -  pvenergytotal        :  95.4
		 -  epvtotal             :  97.3
		 -  epv1today            :  2.5
		 -  epv1total            :  97.3
		 -  epv2today            :  0.0
		 -  epv2total            :  0.0
		 -  pvtemperature        :  18.3
		 -  pvipmtemperature     :  25.1
	 - MQTT jsonmsg: 
			 {"device": "RCL1CHC0CM", "time": "2023-02-19T14:21:57", "buffered": "no",
			 "values": {"datalogserial": "KWK1CH20C5", "pvserial": "RCL1CHC0CM",
			 "pvstatus": 1, "pvpowerin": 3697, "pv1voltage": 2880, "pv1current": 11,
			 "pv1watt": 3696, "pv2voltage": 42, "pv2current": 0, "pv2watt": 1,
			 "pvpowerout": 3625, "pvfrequentie": 4998, "pvgridvoltage": 2334,
			 "pvgridcurrent": 15, "pvgridpower": 3637, "pvgridvoltage2": 0,
			 "pvgridcurrent2": 0, "pvgridpower2": 0, "pvgridvoltage3": 0,
			 "pvgridcurrent3": 0, "pvgridpower3": 0, "totworktime": 2491597,
			 "pvenergytoday": 24, "pvenergytotal": 954, "epvtotal": 973, "epv1today": 25,
			 "epv1total": 973, "epv2today": 0, "epv2total": 0, "pvtemperature": 183,
			 "pvipmtemperature": 251}}
	 - Grott MQTT topic used : energy/growatt
	 - MQTT send failed: [Errno 99] Address not available
	 - Grott Send data to PVOutput disabled 
	 - Grott Send data to Influx disabled 
	 - Grott extension processing started :  grott_ha
	Grott HA 0.0.7-rc5 - creating RCL1CHC0CM config in HA, 32 to push
	Pushing 33 configurations payload to HA
	Configurations pushed
	 - Grott extension processing ended :  0
	 - Growatt packet received:
		  <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('172.30.33.4', 5279), raddr=('192.168.3.142', 1039)>
	 - Data less then minimum record length, data not processed


- Possible Inverter serial TPJ4****
	 - Inverter serial not recognised - using inverter type default
	 - Growatt new layout processing
		 - decrypt       :  True
		 - offset        :  6
		 - record layout :  T06NNNNX
	 - Grott data record date/time processing started
	 - no or no valid time/date found, grott server time will be used (buffer records not sent!)
	 - Grott values retrieved:
		 -  datalogserial        :  KWK1*****
		 -  pvserial             :  TPJ*****
		 -  pvstatus             :  5
		 -  pvpowerin            :  1010.0
		 -  pv1voltage           :  597.8
		 -  pv1current           :  1.1
		 -  pv1watt              :  668.8
		 -  pv2voltage           :  275.7
		 -  pv2current           :  1.2
		 -  pv2watt              :  348.4
		 -  pvpowerout           :  0.0
		 -  pvfrequentie         :  0.0
		 -  pvgridvoltage        :  0.0
		 -  pvgridcurrent        :  0.0
		 -  pvgridpower          :  0.0
		 -  pvgridvoltage2       :  0.0
		 -  pvgridcurrent2       :  0.0
		 -  pvgridpower2         :  0.0
		 -  pvgridvoltage3       :  0.0
		 -  pvgridcurrent3       :  0.0
		 -  pvgridpower3         :  5892186.1
		 -  totworktime          :  54.6
		 -  pvenergytoday        :  0.0
		 -  pvenergytotal        :  0.0
		 -  epvtotal             :  2.0
		 -  epv1today            :  663.5
		 -  epv1total            :  1417597.5
		 -  epv2today            :  5.4
		 -  epv2total            :  337.4
		 -  pvtemperature        :  32.3
		 -  pvipmtemperature     :  215.6
	 - MQTT jsonmsg: 
			 {"device": "TPJ4CEJ07G", "time": "2023-02-19T14:22:02", "buffered": "no",
			 "values": {"datalogserial": "KWK1CH20C5", "pvserial": "TPJ4CEJ07G",
			 "pvstatus": 5, "pvpowerin": 10100, "pv1voltage": 5978, "pv1current": 11,
			 "pv1watt": 6688, "pv2voltage": 2757, "pv2current": 12, "pv2watt": 3484,
			 "pvpowerout": 0, "pvfrequentie": 0, "pvgridvoltage": 0, "pvgridcurrent": 0,
			 "pvgridpower": 0, "pvgridvoltage2": 0, "pvgridcurrent2": 0, "pvgridpower2":
			 0, "pvgridvoltage3": 0, "pvgridcurrent3": 0, "pvgridpower3": 58921861,
			 "totworktime": 393216, "pvenergytoday": 0, "pvenergytotal": 0, "epvtotal":
			 20, "epv1today": 6635, "epv1total": 14175975, "epv2today": 54, "epv2total":
			 3374, "pvtemperature": 323, "pvipmtemperature": 2156}}
	 - Grott MQTT topic used : energy/growatt
	 - MQTT send failed: [Errno 99] Address not available
	 - Grott Send data to PVOutput disabled 
	 - Grott Send data to Influx disabled 
	 - Grott extension processing started :  grott_ha
	Grott HA 0.0.7-rc5 - creating TPJ4CEJ07G config in HA, 32 to push
	Pushing 33 configurations payload to HA
	Configurations pushed
	 - Grott extension processing ended :  0
	 - Growatt packet received:
		  <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('172.30.33.4', 5279), raddr=('192.168.3.142', 1039)>
	 - Data less then minimum record length, data not processed
	 - Growatt packet received:
		  <socket.socket fd=4, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('172.30.33.4', 42244), raddr=('47.91.67.66', 5279)>
	 - Growatt original Data:
		 \x00\x15\x00\x06\x00\xed\x50\x29\x0c\x25\x24\x46\x22\x3c\x46\x77\x31\x5a\x77
		 \x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72
		 \x2e\x2d\x24\x44\x37\x05\x3a\x5f\x4e\x36\x74\x74\x47\x72\x6f\x77\x61\x74\x74
		 \x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x35\x24\x3e\x73\x31\x2a\x3d\x51
		 \x43\x33\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f
		 \x77\x61\x74\x79\xea\x72\x6b\x77\x60\x45\x74\x42\x72\x6e\x46\x61\x72\x74\x45
		 \x41\x5d\x77\x69\x74\x6a\x06\x28\x2a\x47\x22\x36\x3c\x77\x4b\x38\x77\x61\x74
		 \x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x77\x61\x74\x74\x47\x72\x6f\x7e
		 \x61\x75\x44\x47\x78\x6f\x76\x51\x74\x7f\x47\x72\x6f\x7a\x61\x76\x45\x7f\x72
		 \x7b\x77\x69\x33\x20\x14\x25\x5f\x47\x51\x44\x74\x52\x72\x68\x40\x4f\x40\x5a
		 \x77\x5c\x5b\x77\x77\x74\x70\x11\x43\x41\x47\x61\x6f\x74\x46\x44\x6f\x6b\x61
		 \x75\x44\x47\x6c\x6f\x72\x26\x39\x20\x6c\x4a\x6f\x68\x61\x67\x46\x77\x40\x5c
		 \x5a\x51\x46\x59\x76\x4b\x4f\x46\x55\x4e\x46\x76\x48\x5a\x4e\xdb\x2f
	 - Grott automatic protocol detection
	 - Grott data record length 245
	 - layout   :  T065029
	 - no matching record layout found, try generic
	 - Record layout used :  T065029
	 - Growatt data decrypted V2
	 - Grott Growatt data decrypted
	 - Growatt plain data:
		 0015000600ed50294b574b3143483230433500000000000000000000000000000000000000004
		 15a4530434248303957000000000000000000000000000000000000000054504a3443454a3037
		 4700000000000000000000000000000000000000000dad0004000131000500013100060002333
		 20008001e415a4530434248303957000000000000000000000000000000000000000000090001
		 30000a000130000b0000000d0002313800140008475453573030303000150007372e342e302e3
		 40016000456312e30001b000136001c000130001e0005474d542b38001f0013323032332d3032
		 2d31392031343a32313a3539ba5b
	 - Grott data ack record or data record not defined no processing done

**Comment by johanmeijer** [comment]
For the SPH inverter you have to specify invtype=sph in the grott.ini. 

If you have multiple inverters please look at: https://github.com/johanmeijer/grott/wiki/Multi-Inverter-support

Grott will no use layout T06NNNXSPH and that will show also SOC.

**Reply by eismarkt** [reply]
I think now is the Right data there, but I don’t find the soc as entity in ha

1323.0
	 - MQTT jsonmsg: 
			 {"device": "TPJ*****", "time": "2023-02-19T16:34:31", "buffered": "no",
			 "values": {"datalogserial": "KWK*****", "pvserial": "TPJ******",
			 "pvstatus": 5, "pvpowerin": 300, "pv1voltage": 6360, "pv1current": 0,
			 "pv1watt": 299, "pv2voltage": 2914, "pv2current": 0, "pv2watt": 61,
			 "pvpowerout": 2565, "pvfrequentie": 4997, "pvgridvoltage": 4053,
			 "pvgridcurrent": 7, "pvgridpower": 5104, "pvgridvoltage2": 4060,
			 "pvgridcurrent2": 7, "pvgridpower2": 0, "pvgridvoltage3": 3999,
			 "pvgridcurrent3": 7, "pvgridpower3": 0, "totworktime": 14191890, "eactoday":
			 29, "pvenergytoday": 29, "eactotal": 6644, "epvtotal": 4363, "epv1today":
			 60, "epv1total": 3380, "epv2today": 27, "epv2total": 983, "pvtemperature":
			 301, "pvipmtemperature": 293, "pvboosttemp": 289, "bat_dsp": 2124,
			 "eacharge_today": 20, "eacharge_total": 2609, "batterytype": 1,
			 "uwsysworkmode": 5, "systemfaultword0": 0, "systemfaultword1": 0,
			 "systemfaultword2": 0, "systemfaultword3": 0, "systemfaultword4": 0,
			 "systemfaultword5": 0, "systemfaultword6": 0, "systemfaultword7": 0,
			 "pdischarge1": 2359, "p1charge1": 0, "vbat": 2124, "SOC": 64, "pactouserr":
			 0, "pactousertot": 0, "pactogridr": 0, "pactogridtot": 0, "plocaloadr":
			 2600, "plocaloadtot": 2600, "spdspstatus": 5, "spbusvolt": 7135,
			 "etouser_tod": 46, "etouser_tot": 11214, "etogrid_tod": 1, "etogrid_tot":
			 2277, "edischarge1_tod": 13, "edischarge1_tot": 1666, "eharge1_tod": 74,
			 "eharge1_tot": 1887, "elocalload_tod": 76, "elocalload_tot": 13230}}
